rust/lance/src/io/exec/filter.rs

```diff
@@ -0,0 +1,99 @@
+use std::sync::Arc;
+
+use datafusion::{execution::TaskContext, logical_expr::Expr};
+use datafusion_physical_expr::PhysicalExpr;
+use datafusion_physical_plan::{
+    filter::FilterExec, metrics::MetricsSet, DisplayAs, DisplayFormatType, ExecutionPlan,
+    PlanProperties, SendableRecordBatchStream, Statistics,
+};
+use lance_core::{error::DataFusionResult, Result};
+
+#[derive(Debug)]
+pub struct LanceFilterExec {
+    expr: Expr,
+    pub filter: Arc<FilterExec>,
+}
+
+impl DisplayAs for LanceFilterExec {
+    fn fmt_as(&self, t: DisplayFormatType, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        self.filter.fmt_as(t, f)
+    }
+}
+
+// LanceFilterExec is a wrapper around FilterExec that includes the original
+// expression for the filter node. In comparison to a FilterExec, this makes it
+// possible for an optimization rule to serialize the filter to substrait and
+// send it to a remote worker.
+impl LanceFilterExec {
+    pub fn try_new(
+        expr: Expr,
+        predicate: Arc<dyn PhysicalExpr>,
+        input: Arc<dyn ExecutionPlan>,
+    ) -> Result<Self> {
+        let filter_exec = FilterExec::try_new(predicate.clone(), input)?;
+        Ok(Self {
+            expr: expr,
+            filter: Arc::new(filter_exec),
+        })
+    }
+
+    pub fn expr(&self) -> &Expr {
+        &self.expr
+    }
+}
+
+impl ExecutionPlan for LanceFilterExec {
+    fn name(&self) -> &str {
+        "LanceFilterExec"
+    }
+
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn properties(&self) -> &PlanProperties {
+        self.filter.properties()
+    }
+
+    fn children(&self) -> Vec<&Arc<dyn ExecutionPlan>> {
+        self.filter.children()
+    }
+
+    fn maintains_input_order(&self) -> Vec<bool> {
+        self.filter.maintains_input_order()
+    }
+
+    fn with_new_children(
+        self: Arc<Self>,
+        children: Vec<Arc<dyn ExecutionPlan>>,
+    ) -> DataFusionResult<Arc<dyn ExecutionPlan>> {
+        self.filter.clone().with_new_children(children)
+    }
+
+    fn execute(
+        &self,
+        partition: usize,
+        context: Arc<TaskContext>,
+    ) -> DataFusionResult<SendableRecordBatchStream> {
+        self.filter.execute(partition, context)
+    }
+
+    fn metrics(&self) -> Option<MetricsSet> {
+        self.filter.metrics()
+    }
+
+    fn statistics(&self) -> DataFusionResult<Statistics> {
+        self.filter.statistics()
+    }
+
+    fn cardinality_effect(&self) -> datafusion_physical_plan::execution_plan::CardinalityEffect {
+        self.filter.cardinality_effect()
+    }
+
+    fn try_swapping_with_projection(
```

Comment:
Did you make sure that all the methods overridden in `FilterExec` are also overridden here?  Looks like you did, just double checking.

Reply:
Ok.  Generally we don't care too much.  They revolve around optimizations (e.g. calculating statistics correctly for join order or pushing projection up/down) and we don't really rely on Datafusion's optimizer at all.  It tends to be more important for SQL plans but we can work through issues as they arise.